packages/serverless-orchestration/src/ServerlessHub.js

```diff
@@ -520,37 +518,60 @@ async function _getLastQueriedBlockNumber(configIdentifier, chainId, logger) {
 }
 
 function _getWeb3AndUrlForBot(botConfig) {
+  /**
+   * NODE_RETRY CONFIG = [
+   *  { url: "https://...", retries: 2 },  
+   * ]
+   */
+  let urls = [];
   const retryConfig = botConfig?.environmentVariables?.NODE_RETRY_CONFIG;
-  if (retryConfig) {
-    return [new Web3(createBasicProvider(retryConfig)), retryConfig[0].url];
-  } else {
+  if (!retryConfig) {
     const url = botConfig?.environmentVariables?.CUSTOM_NODE_URL || customNodeUrl;
-    return [new Web3(url), url];
+    urls.push({ url, retries: 3 });
```

Comment:
Semi-opinionated default of 3 retries. Should probably be explicitly defined as a constant.